### DISCUSSION TITLE
Curious about disabling cmp keymaps in lazyvim due to a lack of documentation.

### DISCUSSION THREAD

**Discussion by beastr45** [LazyVim/LazyVim#2549] [Q&A]
By default the cmp keymaps are not defined in the keys field so I was curious on how to actually disable some of the default keymaps by merging the config that I introduce.

**Comment by dpetka2001** [comment]
The keymaps for `nvim-cmp` are stored in a table called `mapping`, so you can override that table by doing something like
```lua
return {
  {
    "hrsh7th/nvim-cmp",
    opts = function(_, opts)
      local cmp = require("cmp")
      opts.mapping = vim.tbl_deep_extend("force", opts.mapping, {
        ["<CR>"] = cmp.mapping.confirm({ select = false }),
        -- put here the keymaps that you want to change
      })
    end,
  },
}
```
You should try to learn some basic Lua, because these things can't be documented on a Neovim distro website. If you don't try to teach yourself, you won't be able to make customizations like these. Just basic stuff enough to understand what you're doing, I'm not telling you to become proficient (although you could totally do it if you want it).

**Reply by beastr45** [reply]
thanks I already knew how to do this setting select to false does not solve the issue and dosent disable the enter key either thus the reason I opened a question about disabling cmp keys.

**Reply by dpetka2001** [reply]
Oh you wanted to disable them. I didn't read carefully and thought you wanted to change them to do something else. Then you should do as @abeldekat recommended in his post.

**Reply by Aman9das** [reply]
Thanks

**Comment by abeldekat** [comment]
To remove the CR mapping, set the value of key "CR" to nil:

```lua
{
    "hrsh7th/nvim-cmp",
    opts = function(_, opts)
      opts.mapping["<CR>"] = nil
    end
  }
```

**Comment by beastr45** [comment]
I think I might just overwrite all default settings with my own because for some reason The config is unable to be effective and I think the default config is taking priority.

**Reply by abeldekat** [reply]
That's strange.

Are you able to share the relevant parts of your config, on github for example?

You can also test the code snippet on a fresh LazyVim [starter](https://github.com/LazyVim/starter).
In the steps below, use `NVIM_APPNAME=lazytest nvim` as command to start Neovim.
An alias is convenient: `alias lazytest="NVIM_APPNAME=lazytest nvim"`


1. `mkdir ~/.config/lazytest`
2. `git clone https://github.com/LazyVim/starter ~/.config/lazytest`
3. `lazytest`,  and wait for the installation to complete
4. Restart Neovim: `lazytest`

Now, add the snippet to for example: `~/.config/lazytest/lua/plugins/poc.lua`

Cleanup:

```lua
rm -rf ~/.config/lazytest
rm -rf ~/.local/share/lazytest
rm -rf ~/.local/state/lazytest
rm -rf ~/.cache/lazytest
```

**Reply by beastr45** [reply]
with the defualt config using vim.tbl extend wont change and mapping fromcr to nil. otherwise seems to work. I will try to hammer this into my own config.

**Comment by beastr45** [comment]
Since remapping CR to nil wouldnt want to work with other remappings in my config I just returned the table for cmp twice as a sort of hacky solution and it works so I guess thats okay.
